internal/operators/manager.go

```diff
@@ -69,17 +72,38 @@ func (mgr *Manager) GenerateManifests(cluster *common.Cluster) (map[string]strin
 			manifests, err := operator.GenerateManifests(cluster)
 			if err != nil {
 				mgr.log.Error(fmt.Sprintf("Cannot generate %s manifests due to ", clusterOperator.Name), err)
-				return nil, err
+				return err
 			}
-			if manifests != nil {
-				for k, v := range manifests.Files {
-					operatorManifests[k] = v
+			for k, v := range manifests {
+				err = mgr.createManifests(ctx, cluster, k, v)
+				if err != nil {
+					return err
 				}
 			}
 		}
 	}
 
-	return operatorManifests, nil
+	return nil
+}
+
+func (mgr *Manager) createManifests(ctx context.Context, cluster *common.Cluster, filename string, content []byte) error {
+	// all relevant logs of creating manifest weill be inside CreateClusterManifest
+	response := mgr.manifestsAPI.CreateClusterManifest(ctx, operations.CreateClusterManifestParams{
+		ClusterID: *cluster.ID,
+		CreateManifestParams: &models.CreateManifestParams{
+			Content:  swag.String(base64.StdEncoding.EncodeToString(content)),
+			FileName: &filename,
+			Folder:   swag.String(models.ManifestFolderOpenshift),
+		},
+	})
+
+	if _, ok := response.(*operations.CreateClusterManifestCreated); !ok {
+		if apiErr, ok := response.(*common.ApiErrorResponse); ok {
+			return errors.Wrapf(apiErr, "Failed to create manifest %s", filename)
```

Comment:
pls include cluster id in the message to provide more context to the error.

Reply:
:+1: